File: Data Structures & Algorithms/maximum-frequency-stack/submission-1.cpp
class FreqStack {
    unordered_map<int, int> freq;
    unordered_map<int, stack<int>> stks;
    int maxF;

public:
    FreqStack() {
        maxF = 0;
    }
    
    void push(int val) {
        freq[val]++;
        stks[freq[val]].push(val);
        maxF = max(maxF, freq[val]);
    }
    
    int pop() {
        int ans = stks[maxF].top();
        stks[maxF].pop();
        freq[ans]--;
        if(!stks[maxF].size()) maxF--;
        return ans;
    }
};

/**
 * Your FreqStack object will be instantiated and called as such:
 * FreqStack* obj = new FreqStack();
 * obj->push(val);
 * int param_2 = obj->pop();
 */

File: Data Structures & Algorithms/design-hashset/submission-1.java
class MyHashSet {
    int[] set;
    public MyHashSet() {
       set = new int[31251];
       Arrays.fill(set,0);
    }
    
    public void add(int key) {
        int mask = getMask(key);
        set[key/32] |= (1<<mask);
    }
    
    public void remove(int key) {
        if(contains(key))
            set[key/32] ^= (1<<getMask(key));
    }
    
    public boolean contains(int key) {
        return (set[key/32] & (1<<getMask(key))) != 0;
    }

    private int getMask(int key) {
        return key % 32;
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.add(key);
 * obj.remove(key);
 * boolean param_3 = obj.contains(key);
 */